resharper/resharper-unity/src/Core/Feature/Services/UsageStatistics/UnityExternalFilesFileSizeLogContributor.cs

```diff
@@ -12,33 +14,94 @@ public class UnityExternalFilesFileSizeLogContributor : IActivityLogContributorS
         {
             myUnitySolutionTracker = unitySolutionTracker;
             myAssetIndexingSupport = assetIndexingSupport;
-            myModuleProcessor = moduleProcessor;
         }
 
         public void ProcessSolutionStatistics(JObject log)
         {
             if (!myUnitySolutionTracker.IsUnityProject.HasTrueValue())
                 return;
 
-            if (myModuleProcessor.SceneSizes.Count == 0 && myModuleProcessor.PrefabSizes.Count == 0 && myModuleProcessor.AssetSizes.Count == 0)
+            if (myStatistics.Count == 0)
                 return;
 
+            // TODO: Perhaps we should rethink this?
+            // Does it make more sense to include count, min, max, average instead of all file sizes? This would also
+            // reduce the amount of data we cache here
+
             var stats = new JObject();
-            log["uys"] = stats;
-            stats["s"] = JArray.FromObject(myModuleProcessor.SceneSizes);
-            stats["p"] = JArray.FromObject(myModuleProcessor.PrefabSizes);
-            stats["a"] = JArray.FromObject(myModuleProcessor.AssetSizes);
-            stats["kba"] = JArray.FromObject(myModuleProcessor.KnownBinaryAssetSizes);
-            stats["ebna"] = JArray.FromObject(myModuleProcessor.ExcludedByNameAssetsSizes);
+            log["uys"] = stats; // Unity Yaml Stats
+
+            // User editable files, e.g. contents of Assets and embedded or local packages
+            stats["s"] = JArray.FromObject(GetStatistics(FileType.Scene, true));
+            stats["p"] = JArray.FromObject(GetStatistics(FileType.Prefab, true));
+            stats["a"] = JArray.FromObject(GetStatistics(FileType.Asset, true));
+            stats["ad"] = JArray.FromObject(GetStatistics(FileType.AsmDef, true));
+            stats["kba"] = JArray.FromObject(GetStatistics(FileType.KnownBinary, true));
+            stats["ebna"] = JArray.FromObject(GetStatistics(FileType.ExcludedByName, true));
+
+            // Referenced, or read-only, non-user editable files
+            stats["rs"] = JArray.FromObject(GetStatistics(FileType.Scene, false));
+            stats["rp"] = JArray.FromObject(GetStatistics(FileType.Prefab, false));
+            stats["ra"] = JArray.FromObject(GetStatistics(FileType.Asset, false));
+            stats["rad"] = JArray.FromObject(GetStatistics(FileType.AsmDef, false));
+            stats["rkba"] = JArray.FromObject(GetStatistics(FileType.KnownBinary, false));
+            stats["rebna"] = JArray.FromObject(GetStatistics(FileType.ExcludedByName, false));
+
+            // There's no need to capture all meta file sizes. Just get a count and an average size
+            stats["mfc"] = myStatistics.Count(d => d.FileType == FileType.Meta && d.IsUserEditable);
+            stats["rmfc"] = myStatistics.Count(d => d.FileType == FileType.Meta && !d.IsUserEditable);
+
+            // "All meta file average"
+            stats["amfa"] = myStatistics.Average(d => d.FileType == FileType.Meta ? d.Length : 0.0f);
```

Comment:
Average is not correct, the count is not changed, but some elements contribute 0

Reply:
Oops. Yes, good catch. I'll get it fixed. Are you happy with the statistics that are there?

Reply:
Yes